org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/managers/IBuildSupport.java

```diff
@@ -44,6 +44,14 @@ default boolean isBuildLikeFileName(String fileName) {
 		return false;
 	}
 
+	/**
+	 * Implementors can put preparation work into this method. This method will be invoked before {@code update(...)}.
+	 * All build supports named with 'UnknownBuildTool' will be skipped. See: {@link #buildToolName()}.
```

Comment:
~All build supports named with 'UnknownBuildTool' will be skipped. See: {@link #buildToolName()}.~